src/components/chips/js/chipsDirective.js

```diff
@@ -0,0 +1,159 @@
+	 *   Truncate chip text where possible, but do not truncate entries such that two are indistinguishable.
+	 * Drag and Drop (?)
+	 *   Drag and drop chips between related <md-chips> elements.
+	 *
+	 *
+	 * @param {string=|object=} ng-model A model to bind the list of items to
+	 * @param {string=} placeholder Placeholder text that will be forwarded to the input.
+	 * @param {string=} secondary-placeholder Placeholder text that will be forwarded to the input, displayed when there
+	 *    is at least on item in the list
+	 * @param {boolean=} readonly Disables list manipulation (deleting or adding list items), hiding the input and delete
+	 *    buttons
+	 * @param {expression} md-chip-append An expression expected to convert the input string into an object when adding
+	 *    a chip.
+	 *
+	 * @usage
+	 * <hljs lang="html">
+	 *   <md-chips
+	 *       ng-model="myItems"
+	 *       placeholder="Add an item"
+	 *       readonly="isReadOnly">
+	 *   </md-chips>
+	 * </hljs>
+	 *
+	 */
+
+
+	var MD_CHIPS_TEMPLATE = '\
+			<md-chips-wrap ng-class="{readonly : $mdChipsCtrl.readonly}" class="md-chips">\
+				<div role="presentation">\
+					<div ng-repeat="$chip in $mdChipsCtrl.items"\
+							ng-class="{selected: $mdChipsCtrl.selectedChip == $index}"\
+							ng-click="!$mdChipsCtrl.readonly && $mdChipsCtrl.selectChip($index)"\
+							class="md-chip">\
+					</div>\
+					<div ng-if="!$mdChipsCtrl.readonly" class="md-chip-worker"></div>\
+				</div>\
+			</md-chips-wrap>';
+
+  var CHIP_INPUT_TEMPLATE = '\
+				<input\
+						placeholder="{{$mdChipsCtrl.items.length == 0 ? $mdChipsCtrl.placeholder : $mdChipsCtrl.secondaryPlaceholder}}"\
+						class="md-chip-input"\
+						ng-model="$mdChipsCtrl.chipBuffer"\
+						ng-focus="$mdChipsCtrl.resetSelectedChip()"\
+						ng-keydown="$mdChipsCtrl.defaultInputKeydown($event)">';
+
+  var CHIP_DEFAULT_TEMPLATE = '\
+			<md-chip>\
+				<span>{{$chip}}</span>\
+				<md-button ng-if="!$mdChipsCtrl.readonly"\
+						md-chip-remove>x</md-button>\
+			</md-chip>';
+
+
+  /**
+   * MDChips Directive Definition
+   * @param $mdTheming
+   * @param $log
+   * @ngInject
+   */
+  function MdChips ($mdTheming, $log) {
+    return {
+      template: function(element, attrs) {
+        // Clone the element into an attribute. By prepending the attribute
+        // name with '$', Angular won't write it into the DOM. The cloned
+        // element propagates to the link function via the attrs argument,
+        // where various contained-elements can be consumed.
+        attrs['$mdUserTemplate'] = element.clone();
+        return MD_CHIPS_TEMPLATE;
+      },
+      require: ['ngModel', 'mdChips'],
+      restrict: 'E',
+      controller:   'MdChipsCtrl',
+      controllerAs: '$mdChipsCtrl',
+      bindToController: true,
+      compile: compile,
+      scope: {
+        readonly:             '=readonly',
+        placeholder:          '@',
+        secondaryPlaceholder: '@',
+        mdChipAppend:         '&'
```

Comment:
I don't think so - it's actually meant to be a data-transforming function and not a 'notification' method, so I don't think using an $event is appropriate. I didn't want to have this method in the first place, but haven't figured out a way around it. Specifying this method allows callers to create objects from the input string, so that an object is appended to the list of chips. I'd rather md-chips be used in conjunction with md-autocomplete when a list of objects is desired, as md-autocomplete itself works with lists of objects (or strings).
That being said, I'm not sure yet how "chip editing" is going to play out and whether it will involve this transform as well.
So: I am more than open to suggestions for a new name. md-transform-string-to-object-for-chip-append-expression is too long. And, actually, a misnomer, since callers are able to provide the method even when strings are being used - they could, for example, use it to convert string chips to UPPERCASE..

I've added a comment to the directive indicating its WIP nature and that stuff may break. This officially gives me license to change stuff, like mdChipAppend in the future :)